cvat/apps/dextr_segmentation/dextr.py

```diff
@@ -0,0 +1,112 @@
+
+# Copyright (C) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from openvino.inference_engine import IENetwork, IEPlugin
+
+import os
+import cv2
+import PIL
+import numpy as np
+
+_IE_CPU_EXTENSION = os.getenv("IE_CPU_EXTENSION", "libcpu_extension_avx2.so")
+_IE_PLUGINS_PATH = os.getenv("IE_PLUGINS_PATH", None)
+
+_DEXTR_MODEL_DIR = os.getenv("DEXTR_MODEL_DIR", None)
+_DEXTR_PADDING = 50
+_DEXTR_TRESHOLD = 0.9
+_DEXTR_SIZE = 512
+
+class DEXTR_HANDLER:
+    _plugin = None
+    _network = None
+    _exec_network = None
+    _input_blob = None
+    _output_blob = None
+    _xml = None
+    _bin = None
+
+    def __init__(self):
+        if not _IE_PLUGINS_PATH:
+            raise Exception("IE_PLUGINS_PATH is not defined")
+        if not _DEXTR_MODEL_DIR:
+            raise Exception("DEXTR_MODEL_DIR is not defined")
+        self._xml = os.path.join(_DEXTR_MODEL_DIR, 'dextr.xml')
+        self._bin = os.path.join(_DEXTR_MODEL_DIR, 'dextr.bin')
+
+    def handle(self, im_path, points):
+        # Lazy initialization
+        if not self._plugin:
+            self._plugin = IEPlugin(device="CPU", plugin_dirs=[_IE_PLUGINS_PATH])
+            self._plugin.add_cpu_extension(os.path.join(_IE_PLUGINS_PATH, _IE_CPU_EXTENSION))
```

Comment:
Why not the code below? Can be share inference between auto annotation and the app somehow?

```
        plugin = IEPlugin(device="CPU", plugin_dirs=[IE_PLUGINS_PATH])
        if (self._check_instruction("avx2")):
            plugin.add_cpu_extension(os.path.join(IE_PLUGINS_PATH, "libcpu_extension_avx2.so"))
        elif (self._check_instruction("sse4")):
            plugin.add_cpu_extension(os.path.join(IE_PLUGINS_PATH, "libcpu_extension_sse4.so"))
        else:
            raise Exception("Inference engine requires a support of avx2 or sse4.")
```

Reply:
Yes, I agree
I have done it.